tensorflow/core/common_runtime/eager/BUILD

```diff
@@ -420,16 +420,43 @@ tf_cc_test(
     name = "eager_op_rewrite_registry_test",
     srcs = ["eager_op_rewrite_registry_test.cc"],
     deps = [
-        ":core",
         ":eager_op_rewrite_registry",
-        ":execute",
+        ":mkl_core",
         "//tensorflow/core:lib",
         "//tensorflow/core:no_op_op_lib",
         "//tensorflow/core:test",
         "//tensorflow/core:test_main",
     ],
 )
 
+# Temporary rule until the circular dependencies issue is resolved.
+# TODO(mabuzain): remove this once original "core" package is fixed.
+cc_library(
+    name = "mkl_core",
+    srcs = [
+        "core.cc",
+        "execute.cc",
+        "execute_node.cc",
+    ],
+    hdrs = [
+        "execute.h",
+        "execute_node.h",
+    ],
+    deps = [
+        ":context",
```

Comment:
`//tensorflow/core/common_runtime/eager:eager_op_rewrite_registry_test` is failing to build internally because a number of dependencies from execute.cc are missing. In particular, a few that seem to be missing:

`//absl/container:inlined_vector`
`//absl/types:span`
`//tensorflow/core:framework`
`//tensorflow/core:framework_internal`
`//tensorflow/core:lib`
`//tensorflow/core:lib_internal`

There might be more. But maybe start by copying what's in the ":execute" target.

Reply:
Thanks @jaingaurav. I have fixed that based on your suggestion.

Reply:
Thanks @mahmoud-abuzaina. Seems we are still seeing an internal failure where //tensorflow/core/common_runtime/eager:mkl_core does not depend on a module exporting 'tensorflow/c/c_api_internal.h'.